### PR TITLE
Migrate airflow-azure-data-explorer guide

### LINKED ISSUE #1008: Migrate airflow-azure-data-explorer
Migrate the existing guide airflow-azure-data-explorer (Executing Azure Data Explorer Queries with Airflow) to Docs.

The branch for this is: airflow-azure-data-explorer-migration

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1008 

Migrate the existing guide airflow-azure-data-explorer guide (Executing Azure Data Explorer Queries with Airflow) to Docs.

### CODE REVIEW

**Comment by kwcanuck** [comment]
This PR is ready for review.

**Comment by kentdanas** [comment]
Added an Assumed Knowledge section. Not sure how we want to handle cross-linking guides for the assumed knowledge, since the links will change? 

Otherwise LGTM

**Comment by jwitz** [comment]
@kentdanas I think the best way to link to Airflow guides is to used the assumed internal link (in this case, `what-is-an-operator.md`). Even if we change the title/ filename of this doc later, Docusaurus will be able to detect that the internal link is broken and throw an error

File: guides/airflow-azure-data-explorer.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+> **Note:** All code in this guide can be found on [the Astronomer Registry](https://registry.astronomer.io/dags/azure-data-explorer-tutorial).
+
+## Assumed Knowledge
+
+To get the most out of this guide, you should have knowledge of:
+
+- Airflow operators. See [Operators 101](https://www.astronomer.io/guides/what-is-an-operator/).
+- Running an ADX cluster. See [Quickstart: Create an Azure Data Explorer cluster and database](https://docs.microsoft.com/en-us/azure/data-explorer/create-cluster-database-portal).
+
+## Configuring ADX to Work with Airflow
```
Comment (jwitz) [inline guides/airflow-azure-data-explorer.md:22]:
```suggestion
## Configuring ADX to work with Airflow
```

File: guides/airflow-azure-data-explorer.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+To allow Airflow to communicate with your ADX database, you need to configure service principal authentication. To do this, you create and register an Azure AD service principal, then give that principal permission to access your ADX database. See [Create an Azure Active Directory application registration in Azure Data Explorer](https://docs.microsoft.com/en-us/azure/data-explorer/provision-azure-ad-app).
+
+## Running an ADX Query with Airflow
```
Comment (jwitz) [inline guides/airflow-azure-data-explorer.md:26]:
```suggestion
## Running an ADX query with Airflow
```

File: guides/airflow-azure-data-explorer.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+First, set up an Airflow connection to your ADX cluster. If you do this using the Airflow UI, it should look something like this:
+
+![ADX Connection](/img/guides/adx_connection.png)
+
+The required pieces for the connection are:
+
+- **Host:** your cluster URL
+- **Login:** your client ID
+- **Password:** your client secret
+- **Extra:** should include at least "tenant" with your tenant ID, and "auth_method" with your chosen authentication method. Based on the auth method, you may also need to specify "certificate" and/or "thumbprint" parameters.
```
Comment (jwitz) [inline guides/airflow-azure-data-explorer.md:41]:
```suggestion
- **Host:** Your cluster URL
- **Login:** Your client ID
- **Password:** Your client secret
- **Extra:** Should include at least "tenant" with your tenant ID, and "auth_method" with your chosen authentication method. Based on the auth method, you may also need to specify "certificate" and/or "thumbprint" parameters.
```

File: sidebarsGuides.js [hunk 4]
```diff
@@ -33,8 +33,8 @@ module.exports = {
       label: 'Integrations',
       items: [
         'overview',
-        'airflow-azure-container-instances',
+        'airflow-azure-data-explorer',
```
Comment (jwitz) [inline sidebarsGuides.js:36]:
```suggestion
        'airflow-azure-container-instances',
        'airflow-azure-data-explorer',
```

File: guides/airflow-azure-data-explorer.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+tags: ["Integrations", "Azure", "DAGs"]
+---
+
+[Azure Data Explorer](https://azure.microsoft.com/en-us/services/data-explorer/) (ADX) is a managed data analytics service used for performing real-time analysis of large volumes of streaming data. It's particularly useful for IoT applications, big data logging platforms, and SaaS applications.
+
+You can use the ADX [Hook](https://registry.astronomer.io/providers/microsoft-azure/modules/azuredataexplorerhook) and [Operator](https://registry.astronomer.io/providers/microsoft-azure/modules/azuredataexplorerqueryoperator) which are part of the Azure provider package, to integrate ADX queries into your DAGs. In this guide, you'll learn how to make your ADX cluster work with Airflow and walk through an example DAG that runs a query against a database in that cluster.
+
+> **Note:** All code in this guide can be found on [the Astronomer Registry](https://registry.astronomer.io/dags/azure-data-explorer-tutorial).
+
+## Assumed Knowledge
```
Comment (jwitz) [inline guides/airflow-azure-data-explorer.md:15]:
```suggestion
## Assumed Knowledge
```

**Review by jwitz — APPROVED** [review]

**Pull request merged**